esy-lib/Fs.ml

```diff
@@ -332,7 +342,7 @@ let withTempFile ~data f =
 
   Lwt.finalize
     (fun () -> f (Path.v path))
-    (fun () -> Lwt_unix.unlink path)
+    (fun () -> tryUnlinkLwt path)
```

Comment:
Let's do
```
try%lwt Lwt_unix.unlink path
with Unix_error _ -> Lwt.return
```
inline here. There's no reason to fail on removing a temp file at all.